assets/javascripts/birdeck.js

```diff
@@ -17,32 +22,34 @@ function renderPost(post) {
     + "</p>"
     + "<button id='delete-post' name='button-fetch' class='btn btn-default btn-xs'>Delete</button>"
     + "</div>"
-    )
+  )
+}
+
+function renderPosts(posts) {
+  return posts.map(renderPost)
+}
+
+function addPostToPage(post) {
+  $latestPosts.append(post);
+}
+
+function addPostsToPage(posts) {
+  $latestPosts.html(posts)
+}
+
+function handleError(xhr) {
+  console.log(xhr.responseText)
 }
 
 function fetchPostsButton() {
-  $("#button-fetch-posts").on("click", function(){
-    fetchPosts()
-    })
+  $("#button-fetch-posts").on("click", fetchPosts)
 }
 
 function fetchPosts() {
-  var newestItemID = parseInt($(".post").last().attr("data-id"))
-
-  $.ajax({
-    type:    "GET",
-    url:     "https://turing-birdie.herokuapp.com/api/v1/posts.json",
-    success: function(posts) {
-      $.each(posts, function(index, post) {
-        if (isNaN(newestItemID) || post.id > newestItemID) {
-          renderPost(post)
-        }
-      })
-    },
-    error: function(xhr) {
-      console.log(xhr.responseText)
-    }
-  })
+  $.getJSON("https://turing-birdie.herokuapp.com/api/v1/posts.json")
```

Comment:
@novohispano Cool, let's have them use `.each` exclusively in Ruby as well. Let's also only have them write `.find_by_sql` statements in ActiveRecord too. As it stands, we're having them use a pattern that _no one_ uses and it's not adding to their understanding of anything they come into Module 4.